src/util/bip32.rs

```diff
@@ -167,6 +167,25 @@ impl fmt::Display for ChildNumber {
     }
 }
 
+impl FromStr for ChildNumber {
+    type Err = Error;
+
+    fn from_str(inp: &str) -> Result<ChildNumber, Error> {
+        let mut hardened = false;
```

Comment:
Style nit: can you remove `mut` and `= false`, then add a `hardened = false` to the else clause below?

Reply:
fixed